src/agent/codecoveragepublisher.ts

```diff
@@ -0,0 +1,221 @@
+        var summaryFile = _this.command.properties["summaryfile"];
+
+        _this.readCodeCoverageSummary(summaryFile).then(function(codeCoverageData) {
+            if (codeCoverageData && codeCoverageData.coverageStats && codeCoverageData.coverageStats.length > 0) {
+                _this.executionContext.service.publishCodeCoverageSummary(codeCoverageData, _this.project, _this.buildId)
+                    .then(function(result) {
+                        _this.command.info("PublishCodeCoverageSummary : Code coverage summary published successfully.");
+                        defer.resolve(true);
+                    }).fail(function(error) {
+                        _this.command.warning("PublishCodeCoverageSummary : Error occured while publishing code coverage summary. Error: " + error);
+                        defer.reject(error);
+                    });
+            }
+            else {
+                _this.command.warning("PublishCodeCoverageSummary : No code coverage data found to publish.");
+                defer.resolve(false);
+            }
+        }).fail(function(err) {
+            _this.command.warning("PublishCodeCoverageSummary : Error occured while reading code coverage summary. Error : " + err);
+            defer.reject(err);
+        });
+
+        return defer.promise;
+    }
+    
+    //-----------------------------------------------------
+    // publish code coverage files to server
+    // - reportDirectory: code coverage report directory
+    // - additionalCodeCoverageFiles: additional code coverage files
+    //-----------------------------------------------------
+    public publishCodeCoverageFiles(): Q.Promise<any> {
+        var defer = Q.defer();
+        var _this = this;
+        var containerId = parseInt(_this.executionContext.variables[ctxm.WellKnownVariables.containerId]);
+        var summaryFile = _this.command.properties["summaryfile"];
+        var reportDirectory = _this.command.properties["reportdirectory"];
+        var additionalCodeCoverageFiles = _this.command.properties["additionalcodecoveragefiles"];
+        var codeCoverageArtifactName = "Code Coverage Report_" + _this.buildId;
+        var reportDirectoryExists = false;
+        var newReportDirectory = reportDirectory;
+
+        if (reportDirectory && reportDirectory.length > 0) {
+            if (fs.existsSync(reportDirectory) && fs.lstatSync(reportDirectory).isDirectory()) {
+                reportDirectoryExists = true;
+            }
+            else {
+                _this.command.warning("Report directory '" + reportDirectory + "' doesnot exist or it is not a directory.");
+            }
+        }
+
+        if (!reportDirectoryExists) {
+            newReportDirectory = path.join(shell.tempdir(), "CodeCoverageReport_" + _this.buildId);
+            shell.mkdir('-p', newReportDirectory);
+        }
+      
+        // copy the summary file into report directory
+        shell.cp('-f', summaryFile, newReportDirectory);
```

Comment:
If report directory is not there we give a warning and go ahead. 
Check for summary file is there in coedcoverage.publish . If summary file is not present we will fail the task.